src/test/java/org/broadinstitute/hellbender/tools/FixCallSetSampleOrderingIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+        headerLines.add(formatField);
+        headerLines.add(VCFStandardHeaderLines.getFormatLine("GT"));
+
+        for( int i = 0; i < howMany ; i++) {
+            final File out = File.createTempFile("tiny_", ".vcf");
+
+            try (final VariantContextWriter writer = GATKVariantContextUtils.createVCFWriter(out, dict, false, Options.INDEX_ON_THE_FLY)) {
+                final String sampleName = "Sample_" + String.valueOf(i);
+
+                final VCFHeader vcfHeader = new VCFHeader(headerLines, Collections.singleton(sampleName));
+                vcfHeader.setSequenceDictionary(dict);
+                writer.writeHeader(vcfHeader);
+                final Allele Aref = Allele.create("A", true);
+                final Allele C = Allele.create("C");
+                final List<Allele> alleles = Arrays.asList(Aref, C);
+                final VariantContext variant = new VariantContextBuilder("invented", contig, 100, 100, alleles)
+                        .genotypes(new GenotypeBuilder(sampleName, alleles).attribute(SAMPLE_NAME_KEY, sampleName).make())
+                        .make();
+                writer.add(variant);
+                nameToFileMap.put(sampleName, out);
+            }
+        }
+
+        return IOUtils.writeTempFile(
+                nameToFileMap.entrySet().stream()
+                        .map(pair -> pair.getKey() + "\t" + pair.getValue())
+                        .sorted(Comparator.reverseOrder())
+                        .collect(Collectors.joining("\n")), "mapping", ".sample_map");
+    }
+
+    /**
+     * this was used to generate the vcf files files used in {@link #testUnshufflingRestoresCorrectSamples(File, File, int)}
+     * it's kept around in case we need to generate new test cases or examine how they were made
+     */
+    @Test(enabled = false)
+    public void generateVCFs() throws IOException {
+        final File file = writeManyVCFs(1000);
+        System.out.println(file);
+    }
+
+    /**
+     *   test files were created by generating mangled files using the {@link #writeManyVCFs(int)} with parameter 1000
+     *   these were then imported to GenomicsDB using GATK 4.beta.5-66-g9609cb3-SNAPSHOT which is before the fix for GenomicsDBImport
+     *   sample name ordering was applied
+     *
+     *   the combined vcfs were then extracted using SelectVariants, these are the badlySorted1000 files
+     *   each vcf contains a single record with 1000 genotypes that have a single format field "SN: sample name"
+     *   these were created with SN's that matched their assigned sample names, but the badly sorted versions are shuffled
+     */
+    @DataProvider
+    public Object[][] getBadlySortedFiles(){
+        return new Object[][]{
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile(BADLY_SORTED1000_BATCH_SIZE50_VCF), 50},
+                {getTestFile(REVERSE_ORDERED_SAMPLE_MAP), getTestFile("badlySorted1000-batch-size50.vcf.gz"), 50},
```

Comment:
Can you confirm that given a `vcf.gz` input, the tool writes a `vcf.gz` output?

Reply:
it writes a gz if you ask it to, and produces an index either way, unless you ask it not to